_posts/2023-03-09-prospero.adoc

```diff
@@ -0,0 +1,293 @@
+  - id: "jboss-public"
+    url: "https://repository.jboss.org/nexus/content/groups/public/"
+manifest:
+  url: "file:examples/wildfly-27.0.0.Alpha2-manifest.yaml"
+----
+
+The above channel file defines the repositories to be used for downloading components. In addition, it includes a manifest file:
+
+[,yaml]
+----
+manifest:
+  url: "file:examples/wildfly-27.0.0.Alpha2-manifest.yaml"
+----
+
+Here is part of the contents of `examples/wildfly-27.0.0.Alpha2-manifest.yaml`:
+
+[,yaml]
+----
+➤ head -n 20 examples/wildfly-27.0.0.Alpha2-manifest.yaml
+---
+schemaVersion: "1.0.0"
+name: Channel for WildFly 27
+description: |-
+  This channel provides updates WildFly 26 Feature Pack.
+streams:
+  - groupId: "org.wildfly"
+    artifactId: "wildfly-galleon-pack"
+    version: "27.0.0.Alpha2"
+
+  - groupId: "org.wildfly"
+    artifactId: "wildfly-ee-galleon-pack"
+    version: "27.0.0.Alpha2"
+
+  - groupId: "org.wildfly.core"
+    artifactId: "wildfly-core-galleon-pack"
+    version: "19.0.0.Beta13"
+
+  - groupId: "org.wildfly.core"
+    artifactId: "wildfly-version"
+----
+
+As the command output shown above, the above manifest file defines the component versions. Combining the repositories and the component versions, the `prospero` tool knows how to provision a WildFly server, and the component versions in the WildFly installation is managed by the tool. Next we'll see how to use the `prospero` to update or rollback the provisioned WildFly distribution.
+
+== Other Usages Of The Tool
+
+The `prospero` tool itself contains help to its usage:
+
+[,bash]
+----
+➤ ./prospero
+Welcome to prospero CLI!
+
+This tool allows you to provision instances of the Wildfly application container.
+
+Usage: prospero [-hv] [COMMAND]
+
+Options:
+  -h, --help      Display this help message.
+  -v, --version   Print prospero version and exit.
+
+Commands:
+  install         Install a new application server instance.
+  update          Updates a server instance with the latest patches.
+  print-licences  Print any licences and additional agreements required to install chosen server.
+  history         List previous installation states.
+  revert          Reverts to a previous installation state.
+  channel         Manage channels the installation is subscribed to.
+  completion      Generate bash completion script. (Apply with `source <(prospero completion)`)
+  clone           Manage the installation metadata.
+
+Exit codes:
+  0    Successful program execution, no changes made to the installation.
+  1    Successful program execution, installation files were changed.
+  -1   Invalid arguments were given.
+  -2   Failed operation.
+
+Use `prospero <COMMAND> --help` to show usage information about a given command.
+----
+
+In the above command output, it has a list of the commands supported. Firstly we can try to use its `update` command. We can update one of the component versions defined in the manifest file `examples/wildfly-27.0.0.Alpha2-manifest.yaml`:
```

Comment:
Just wanted to drop here a note discussed a bit in the latest provisioning meeting; we should be clear that updating a component by manually editing the manifest is under user responsibility if the changes of this manifest don't come from an "official" manifest, for example, one manifest generated from a more recent version, you could break your server installation.

Reply:
okay! I'll add this note into the article :D